--- openmpi-2.0.2/orte/tools/orterun/orterun.c	2017-03-22 10:23:01.668834000 -0700
+++ ompi/orte/tools/orterun/orterun.c	2017-03-22 10:25:13.824574000 -0700
@@ -2383,6 +2383,22 @@
         opal_setenv(env_name, "1", true, &app->env);
     }
     free(env_name);
+    /* setup the attach fifo in case someone wants to re-attach */
+    if (orte_create_session_dirs) {
+        /* create the attachment FIFO and setup readevent - cannot be
+         * done if no session dirs exist!
+         */
+        attach_fifo = opal_os_path(false, orte_process_info.job_session_dir,
+                                   "debugger_attach_fifo", NULL);
+        if ((mkfifo(attach_fifo, FILE_MODE) < 0) && errno != EEXIST) {
+            opal_output(0, "CANNOT CREATE FIFO %s: errno %d", attach_fifo, errno);
+            free(attach_fifo);
+            return;
+        }
+        strncpy(MPIR_attach_fifo, attach_fifo, MPIR_MAX_PATH_LENGTH - 1);
+        free(attach_fifo);
+        open_fifo();
+    }
 }

 static void setup_debugger_job(void)
